File: 80.py
from typing import List


def removeDuplicates(nums: List[int]) -> int:
    found = 1
    is_twice = False
    last = nums[0]
    for i in range(1, len(nums)):
        if nums[i] == last:
            if not is_twice:
                is_twice = True
                nums[found] = nums[i]
                found+=1
                continue
        else:
            is_twice = False
            nums[found] = nums[i]
            found+=1
            last = nums[i]
    return found

if __name__=="__main__":
    a = [0, 0, 1, 1, 1, 1, 2, 3, 3]
    print(removeDuplicates(a), a)

File: 80.cpp

#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    int l = nums[0];
    int seen = 0;
    int k = 1;
    for (size_t i = 1; i < nums.size(); ++i) {
      if (nums[i] == l) {
        if (!seen) {
          seen = 1;
          nums[k] = nums[i];
          k++;
        }
      } else {
        seen = 0;
        l = nums[i];
        nums[k] = nums[i];
        k++;
      }
    }
    return k;
  }
};

int main() {
  vector<int> v = {0,0,1,1,1,1,2,3,3};
  Solution s;
  cout << s.removeDuplicates(v) << endl;
  for (const auto &x : v) {
    cout << x << " ";
  }
  cout << endl;
}